controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -0,0 +1,99 @@
+package clusterurlmonitor
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/blackboxexporter"
+	"github.com/openshift/route-monitor-operator/pkg/const/blackbox"
+	utilfinalizer "github.com/openshift/route-monitor-operator/pkg/util/finalizer"
+	"github.com/openshift/route-monitor-operator/pkg/util/reconcile"
+	"github.com/openshift/route-monitor-operator/pkg/util/templates"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ClusterUrlMonitorSupplement struct {
```

Comment:
if you created a different struct, you don't need to be in the same package as the reconciler, and the supplement can be used as a package

Reply:
what's the advantage of moving a single struct into its own package? As long as the only user is the `clusterurlmonitor` package I would prefer to keep it there.